packages/jest-cli/src/cli/runCLI.js

```diff
@@ -45,6 +45,17 @@ module.exports = (
     return;
   }
 
+  const config = readConfig(argv, root);
+  if (argv.showConfig) {
+    const output = ({config}) => {
+      pipe.write(JSON.stringify(config, null, 2));
```

Comment:
Based on what @cpojer [mentioned](https://github.com/facebook/jest/issues/3288#issuecomment-293636469) I think we can make this use [`logDebugMessage`](https://github.com/facebook/jest/pull/3296/files#diff-8006e39142c24851115752df17e9c1c0R61) and update `logDebugMesage` to print out a JSON... Is that ok @cpojer?